File: 1236-n-th-tribonacci-number/n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {

        if(n== 0)
         return 0;
         if(n== 1 || n==2)
         {
            return 1;
         }

        int t1 =0;
        int t2 = 1;
        int t3 = 1;
        int d = 0;

        for(int i=3; i<=n;i++)
        {
             d = t1+t2+t3;
            t1 = t2;
            t2 = t3;
            t3 = d;
        }
        return d;
    }
}

File: 1236-n-th-tribonacci-number/n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {

        if(n < 3)
        {
            return n > 0 ? 1:0;
        }

        vector<int>dp(n+1,0);
        dp[1] = 1;
        dp[2] = 1;

        for(int i= 3 ; i<=n ; i++)
        {
            dp[i] = dp[i-1]+ dp[i-2]+dp[i-3];
        }
        return dp[n];
    }
};